### ISSUE TITLE
The use of HtmlWidget in IOS 16 causes scrolling lag when the HTML content contains numerous anchor tags.

### ISSUE THREAD

**Issue by bksubhuti** [daohoangson/flutter_widget_from_html#875]
## Steps to Reproduce
See attached zip

We are getting complaints with ios 16 for scroll lag.
The scroll is laggy in debug but tolerable in release mode.  We have an app with about 2000 users all os's
Maybe 300 in Apple

The issue is cauesd by our function called _makeClickable  If you comment out the code and just put
    html = _makeClickable(content);

// can replace with this
//html = content; 

then it will work better.

I don't have ios to challenge this.
It is an ios 16 bug and not for ios 15

The main programmer says it is a bug from Html widget and he made a vanilla app to show and the anchors are the cause.

This code necessary to have single click dictionary lookup in our app.
While we wanted to have  you implement a feature **clickable** (still waiting) , we might still need to make anchors our own method for being able to detect which word was actually clicked (if the word happens more than one time in a page, then we cannot know which one).  We have not implemented a method, but we know how to do it by encoding the position into the href.

## Expected results
expect tolerable scrolling and workable lag.  
You will only see a big problem with ios 16 ipad devices.
Our researched issue.
https://github.com/bksubhuti/tipitaka-pali-reader/issues/145

[sample_bug.zip](https://github.com/daohoangson/flutter_widget_from_html/files/10948543/sample_bug.zip)

here is regex load time per page, which is insignificant.

![image](https://user-images.githubusercontent.com/18184280/224485643-54796d26-ca99-4d2c-9df5-9307eab9478e.png)

**Comment by bksubhuti** [comment]
There was some confusion with side bar scroll or just plain scroll by dragging the list view.
The video shows ios16 which has the problem .
I am relaying this bug and video for our main developer who has access to ios simulator.  **It does not happen in ios 15**.  If you want he can downgrade to make a video, but, please request it.  It is a big job.
We think it is related to the gesture detectors in the usage of the hfref links.  if there are 10,000 words, then we have 10,000 gesture detectors because each word is a link.  We do this so we can have single click word lookup.  If you implement something different let us know.

https://user-images.githubusercontent.com/18184280/224701150-b668f087-37f4-44f0-b6bd-d778bb45623d.mp4

**Comment by bksubhuti** [comment]
Are you able to reproduce it with the sample app given as an attachment?
It will work in ios15 but not 16

**Comment by daohoangson** [comment]
You have this massive number of recognizers because you need single tap to select right? This issue https://github.com/daohoangson/flutter_widget_from_html/issues/673?

**Comment by bksubhuti** [comment]
Yes.  We make everything secretly  an a href .
But it appears this bug is independent from your html widget.  It is based on adding gesture detectors.  

We tried to place an issue with flutter on this but since we don't have access to a real ipad device we cannot provide stats.  We have a vanilla app we submitted that reproduces on an emulator but it would not be accepted and was closed.  It seems to happen in 16.x iOS only.
